app/src/main/java/com/mapbox/navigation/examples/replay/ReplayHistoryActivity.kt

```diff
@@ -0,0 +1,368 @@
+import com.mapbox.maps.plugin.LocationPuck2D
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.listeners.OnMapLoadErrorListener
+import com.mapbox.maps.plugin.locationcomponent.LocationComponentPlugin
+import com.mapbox.maps.plugin.locationcomponent.OnIndicatorPositionChangedListener
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.base.options.NavigationOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.MapboxNavigationProvider
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.replay.MapboxReplayer
+import com.mapbox.navigation.core.replay.history.ReplayEventBase
+import com.mapbox.navigation.core.replay.history.ReplaySetNavigationRoute
+import com.mapbox.navigation.core.trip.session.LocationMatcherResult
+import com.mapbox.navigation.core.trip.session.LocationObserver
+import com.mapbox.navigation.core.trip.session.RouteProgressObserver
+import com.mapbox.navigation.examples.R
+import com.mapbox.navigation.examples.databinding.ActivityReplayHistoryLayoutBinding
+import com.mapbox.navigation.ui.maps.NavigationStyles
+import com.mapbox.navigation.ui.maps.camera.NavigationCamera
+import com.mapbox.navigation.ui.maps.camera.data.MapboxNavigationViewportDataSource
+import com.mapbox.navigation.ui.maps.camera.transition.NavigationCameraTransitionOptions
+import com.mapbox.navigation.ui.maps.location.NavigationLocationProvider
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineApi
+import com.mapbox.navigation.ui.maps.route.line.api.MapboxRouteLineView
+import com.mapbox.navigation.ui.maps.route.line.model.MapboxRouteLineOptions
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineColorResources
+import com.mapbox.navigation.ui.maps.route.line.model.RouteLineResources
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+
+private const val DEFAULT_INITIAL_ZOOM = 15.0
+
+class ReplayHistoryActivity : AppCompatActivity() {
+
+    private var loadNavigationJob: Job? = null
+    private val navigationLocationProvider = NavigationLocationProvider()
+    private lateinit var historyFileLoader: HistoryFileLoader
+    private lateinit var mapboxNavigation: MapboxNavigation
+    private lateinit var mapboxReplayer: MapboxReplayer
+    private lateinit var locationComponent: LocationComponentPlugin
+    private lateinit var navigationCamera: NavigationCamera
+    private lateinit var viewportDataSource: MapboxNavigationViewportDataSource
+    private lateinit var binding: ActivityReplayHistoryLayoutBinding
+    private lateinit var activityResultLauncher: ActivityResultLauncher<Intent>
+    private var isLocationInitialized = false
+    private val pixelDensity = Resources.getSystem().displayMetrics.density
+    private val overviewPadding: EdgeInsets by lazy {
```

Comment:
let's simplify a bit example: remove `EdgeInsets`s and lock the screen to portrait only. This way we remove code that is not directly related to `Replay`